lib/fxpay/settings.js

```diff
@@ -102,22 +112,43 @@
         exports.payProviderUrls[paySpec] = exports.extraProviderUrls[paySpec];
       }
     }
+    if (exports.fakeProducts) {
+      exports.allowTestReceipts = true;
+    }
 
+    // Construct our in-app payments adapter.
     var DefaultAdapter = fxpay.getattr('adapter').FxInappAdapter;
     if (!exports.adapter) {
-      exports.log.info('using default adapter');
+      exports.log.info('creating default adapter');
       exports.adapter = new DefaultAdapter();
-    } else {
-      exports.log.info('using custom adapter');
     }
 
-    if (exports.fakeProducts) {
-      exports.allowTestReceipts = true;
-    }
+    // Configure the new adapter or re-configure an existing adapter.
+    exports.adapter.configure(exports);
+    exports.log.info('using adapter:', exports.adapter.toString());
 
-    exports.log.info('fxpay version:', exports.libVersion);
+    exports.log.info('(re)configuration completed; fxpay version:',
+                     exports.libVersion);
+    exports.alreadyConfigured = true;
 
     return exports;
   };
 
+
+  exports.initialize = function(newSettings) {
+    //
+    // A hook to ensure that settings have been initialized.
+    // Any public fxpay method that a user may call should call
+    // this at the top. It can be called repeatedly without harm.
```

Comment:
heh, yes, I thought about using that word but 'can be called repeatedly without harm' also conveys it in only a few more words.